Library/Homebrew/cask/cask.rb

```diff
@@ -440,6 +440,8 @@ def to_hash_with_variations(hash_method: :to_h)
           MacOSVersion::SYMBOLS.keys.product(OnSystem::ARCH_OPTIONS).each do |os, arch|
             bottle_tag = ::Utils::Bottles::Tag.new(system: os, arch:)
             next unless bottle_tag.valid_combination?
+            next if depends_on.macos && !bottle_tag.to_macos_version.compare(depends_on.macos.comparator,
+                                                                             depends_on.macos.version)
 
             Homebrew::SimulateSystem.with(os:, arch:) do
```

Comment:
In theory calling `next if depends_on.macos && !depends_on.macos.satisfied?` here would check if the macOS requirement is satisfied for the simulated OS, but `satisfied?` is still checking the actual system version and its code is beyond my current capabilities.

Reply:
Agreed. Maybe @reitermarkus has ideas here.